packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -43,9 +63,31 @@ const decorator = (service) => ({
     const data = { ...opts.data };
     if (isNull(data[ENTITY_STAGE_ATTRIBUTE])) {
       delete data[ENTITY_STAGE_ATTRIBUTE];
+      return service.update.call(this, uid, entityId, { ...opts, data });
+    }
+
+    const previousStage = await getEntityStage(uid, entityId);
+
+    const updatedEntity = await service.update.call(this, uid, entityId, { ...opts, data });
+    if (
+      previousStage?.workflow?.id &&
```

Comment:
is it possible a stage does not belong to any workflow?

Reply:
no but previous stage could return as `{}` if the stage attribute is not found on the entity